src/modules/adaptors/Frax/DebtFTokenAdaptor.sol

```diff
@@ -0,0 +1,310 @@
+        FRAX = ERC20(_frax);
+    }
+
+    //============================================ Global Functions ===========================================
+    /**
+     * @dev Identifier unique to this adaptor for a shared registry.
+     * Normally the identifier would just be the address of this contract, but this
+     * Identifier is needed during Cellar Delegate Call Operations, so getting the address
+     * of the adaptor is more difficult.
+     */
+    function identifier() public pure virtual override returns (bytes32) {
+        return keccak256(abi.encode("FraxLend debtToken Adaptor V 1.0"));
+    }
+
+    //============================================ Implement Base Functions ===========================================
+
+    /**
+     * @notice User deposits are NOT allowed into this position.
+     */
+    function deposit(
+        uint256,
+        bytes memory,
+        bytes memory
+    ) public pure override {
+        revert BaseAdaptor__UserDepositsNotAllowed();
+    }
+
+    /**
+     * @notice User withdraws are NOT allowed from this position.
+     */
+    function withdraw(
+        uint256,
+        address,
+        bytes memory,
+        bytes memory
+    ) public pure override {
+        revert BaseAdaptor__UserWithdrawsNotAllowed();
+    }
+
+    /**
+     * @notice This position is a debt position, and user withdraws are not allowed so
+     *         this position must return 0 for withdrawableFrom.
+     */
+    function withdrawableFrom(bytes memory, bytes memory) public pure override returns (uint256) {
+        return 0;
+    }
+
+    /**
+     * @notice Returns the cellar's balance of the respective Fraxlend debtToken calculated from cellar borrow shares
+     * @param adaptorData encoded fToken (fraxlendPair) for this position
+     * TODO: CRISPY QUESTION - should we call them fTokens, or fraxlendPairs? I've gone ahead and made them fTokens to stay consistent, but it could be argued that for lending/borrowing it makes more sense to specify it as FraxlendPair.
+     * TODO: CRISPY QUESTION - From looking at AaveDebtTokenAdaptor balanceOf() it looks like it just reports the amount of debtTokens it has, not the overall owed amount? As in, if the cellar borrowed a bunch of DAI from Aave, and then used it, it wouldn't account for it anymore. Need to think on this more. I'd think we need to show the amount of debt that this cellar has/is accruing (thus account for interest would be set to true).
+     */
+    function balanceOf(bytes memory adaptorData) public view override returns (uint256) {
+        IFToken fToken = abi.decode(adaptorData, (IFToken));
+        return _toBorrowAmount(fToken, fToken.userBorrowShares(msg.sender), false, ACCOUNT_FOR_INTEREST);
+    }
+
+    /**
+     * @notice Returns FRAX.
+     */
+    function assetOf(bytes memory) public view override returns (ERC20) {
+        return FRAX;
```

Comment:
Could rename this to debtToken or something since in the future it could be something different

Reply:
I think I made a comment on this but:

- Community from FRAX seems to think that they will not ever have a different `debtAsset`

That said, we'll make it the return var from `FraxlendPair.asset()`